### ISSUE TITLE
VS Code: Auto env activation fails due to faulty usage of "conda activate" and "conda run"

### ISSUE THREAD

**Issue by Zaubeerer** [prefix-dev/pixi#412]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example
Install VS Code: https://code.visualstudio.com/download

`pixi init`, `pixi add python=3.10 pytest` a simplistic pixi project such as:
```shell
[project]
name = "simplistic_project"
version = "0.1.0"
description = "Add a short description here"
authors = ["zaubeerer"]
channels = ["conda-forge", "pytorch", "nvidia"]
platforms = ["linux-64"]

[tasks]

[dependencies]
python = "3.10.*"
pytest
```

Create a simplistic `test_nothing.py`:

```python
def test_pass():
   pass
```

Manually select the pixi python executable as Python interpreter as described in https://github.com/prefix-dev/pixi/issues/166#issuecomment-1616602232, if it does not work automatically (see #411).

Refresh test discovery in the Testing Extension, which should fail with an error.

### Issue description

As described in #411 and [discussed on Discord Server](https://discordapp.com/channels/1082332781146800168/1082338253925003385/1166398225259958422), even after manual selection of the pixi python executable as Python interpreter, the tests are not discovered. 

The reason for the malfunction is that VS Code attempts to activate the environment via `conda activate ...`, thus leading to an error as follows:

```sh
Error: Command failed: conda run -p {project_path}/.pixi/env
(...)
DirectoryNotACondaEnvironmentError: The target directory exists, but it is not a conda environment.
```

### Expected behavior

VS Code should activate the pixi environment without using conda for:
- creation of internal terminals
- pytest test discovery in the Testing Extension

### Further Information
- [ChatGPT Conversation](https://chat.openai.com/share/73ef237d-0ec4-46b8-9957-fe520a2496a7): I tried all suggestions, without success.

### IDE Information
I encountered this error on Microsoft WSL, using the following VS Code version:

Version: 1.83.1 (user setup)
Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc
Date: 2023-10-10T23:48:05.904Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.23570

**Comment by Zaubeerer** [comment]
@pavelzw, did you already find a solution for this error?

**Comment by Zaubeerer** [comment]
From a VS Code user perspective, I would say that fixing this issue is a **must-have**, while #411 and #166 are **nice-to-have**.

I will watch this issue and reconsider using pixi, when there is a workaround or solution for this.

**Comment by ruben-arts** [comment]
Hello @Zaubeerer, unfortunately am I unable to recreate your issue. For me I only had to do the things @pavelzw points out in #166 Which you clearly found as well. The `conda run` part is new to me. So I have a few questions:
- Would it be possible you have some kind of conda extension for vscode installed? 
- Do you have a conda base environment active?
- Which version of vs-code do you use?
I hope we can figure it out!

**Comment by Zaubeerer** [comment]
That is great news!

Thanks for following up so quickly 🙏 

Will test check for those options again later today and let you know.

**Comment by Zaubeerer** [comment]
Hey @ruben-arts, I uninstalled all conda, mamba, micromamba and miniconda and now I don't get the `conda run` anymore.

However, now I get the following error:

```
2023-10-27 16:32:15.582 [info] Discover tests for workspace name: pixi-hello-world - uri: [...]/pixi/pixi-hello-world
2023-10-27 16:32:15.583 [info] Python interpreter path: ./.pixi/env/bin/python3.12
2023-10-27 16:32:15.637 [info] > ./.pixi/env/bin/python3.12 ~/.vscode-server/extensions/ms-python.python-2023.18.0/pythonFiles/testing_tools/run_adapter.py discover pytest -- --rootdir . -s --cache-clear .
2023-10-27 16:32:15.638 [info] cwd: .
2023-10-27 16:32:15.639 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
```

**Comment by Zaubeerer** [comment]
I just switched from `PyLance` to `Jedi` as language server.

However, I got a similar error:

```
2023-10-27 16:38:25.344 [info] > ./.pixi/env/bin/python3.12 -I ~/.vscode-server/extensions/ms-python.python-2023.18.0/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2023.18.0/pythonFiles/interpreterInfo.py
2023-10-27 16:38:25.381 [info] Python interpreter path: ./.pixi/env/bin/python3.12
2023-10-27 16:38:25.426 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-10-27 16:38:25.426 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-10-27 16:38:25.426 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2023-10-27 16:38:26.278 [info] Starting Jedi language server for pixi-hello-world.
2023-10-27 16:38:26.286 [error] Active interpreter type is detected as Unknown {"id":"/home/zaubeerer/code/pdm/podcasts/pixi/pixi-hello-world/.pixi/env/bin/python3.12","sysPrefix":"/home/zaubeerer/code/pdm/podcasts/pixi/pixi-hello-world/.pixi/env","envType":"Unknown","envName":"","envPath":"/home/zaubeerer/code/pdm/podcasts/pixi/pixi-hello-world/.pixi/env","path":"/home/zaubeerer/code/pdm/podcasts/pixi/pixi-hello-world/.pixi/env/bin/python3.12","architecture":3,"sysVersion":"3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:43:22) [GCC 12.3.0]","version":{"raw":"3.12.0","major":3,"minor":12,"patch":0,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.12.0 64-bit","detailedDisplayName":"Python 3.12.0 64-bit","type":"Conda"}
```

**Comment by Zaubeerer** [comment]
PS: Running `pytest .` in an activated `pixi shell` works fine 🤔

**Comment by Zaubeerer** [comment]
I removed all `conda`, `mamba`, `micromamba` and `miniconda` and now it works :) 

Hence, for now I am no longer blocked.

However, I suggest to keep this issue open, because conda and pixi should be installable in parallel in my opinion.

I will check, whether I can reinstall `conda`, `mamba`, etc. without reintroducing the error.

**Comment by Zaubeerer** [comment]
Installation of micromamba seemed not to introduce the buggy behavior I experienced before.

I suggest to observe it for a while.

At this point, it would seem to me that compatibility with micromamba would be sufficient

**Comment by Zaubeerer** [comment]
@ruben-arts, shall we somehow document this?

Happy to contribute a small PR if you think that makes sense :)

**Comment by ruben-arts** [comment]
We always love contributions, though I'm not sure what would be the proper place for this. 

I changed the title of this issue so it's easier to search for. Do you think that would be enough for now?

**Comment by Zaubeerer** [comment]
Yeah, I would probably have found it given the VS Code in the name. :)
